amqp/connection.py

```diff
@@ -569,12 +569,14 @@ def close(self, reply_code=0, reply_text='', method_sig=(0, 0),
             return
 
         try:
+            self.is_closing = True
```

Comment:
What happens if you call [Connection.collect](https://github.com/celery/py-amqp/pull/220/files#diff-2a6a0da3750a5417c3a6341ad912f551R452) here, instead of handling this when `Channel._do_revive` is called?

Reply:
It cannot be done. Client must wait for CloseOk method and after do the clean up - see the close method docs [1]

[1] https://www.rabbitmq.com/amqp-0-9-1-reference.html